Test improvements

mistydemeo opened the pull request:
- List test deps in requirements/test.txt
- Set null loggers for the root logger, instead of using the default loggers, so that the tests can be run without having a `/var/archivematica/sharedDirectory`
- Update python-keystoneclient, which allows Swift tests to pass

Commit:
Move test deps to requirements/test.txt

Commit:
settings/test: all loggers use console

Commit:
Update python-keystoneclient

Hwesta commented on storage_service/storage_service/settings/test.py:
```diff
@@ -13,3 +13,23 @@
+        }
+    },
+    'handlers': {
+        'null': {
```
Have a console logger instead of a null logger so we can still see debugging output?  Does py.test still print logging output if the test fails with it configured like this?

mistydemeo replied:
Console logger might be a good idea, yeah. I was assuming that tests were likely to raise, not silently fail for other reasons, but that might not be a fair assumption.

Hwesta replied:
And it's useful for troubleshooting why the test failed.

mistydemeo replied:
Switched to using `StreamHandler` at the default of stderr.

Commit:
Tests: fix working-directory sensitive content

* Load cassettes from absolute, not relative, path
* Reference other fixtures at absolute path

mistydemeo commented:
Added a commit which fixes working-directory sensitive things, like VCR cassette locations. This makes it possible to run the tests from the repository root.

Hwesta commented on storage_service/storage_service/settings/test.py:
```diff
@@ -13,3 +13,29 @@
     },
 }
+
+LOGGING = {
```
Update `pytest.ini` to point at this not settings.local?

mistydemeo replied:
Done!

Commit:
pytest.ini: use settings.test

Commit:
test_models: remove failing test

Commit:
test_models: use VCR for remaining test

Commit:
Rename test_models to test_lockssomatic

Commit:
Remove pytest-pythonpath

This path might not actually be the path of the Storage Service being
tested; having this PYTHONPATH set makes it impossible to test an
installation that isn't installed into /usr/lib.

Hwesta committed:
Tests: Fix startup tests for import problems

startup runs on import, so move the import inside the TestCase (after the
database has been set up), and clear the DB explicitly after the extra run.

Hwesta commented:
As discussed, this reveals a problem with `test_startup.py` where the tested function is imported before setup finished, and is run on import.  Pushed a suggested fix - what do you think?

Hwesta commented:
:+1:
